test/integration/private_zonal_with_networking/private_zonal_with_networking_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package private_zonal_with_networking
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/golden"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPrivateZonalWithNetworking(t *testing.T) {
+	bpt := tft.NewTFBlueprintTest(t)
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		// Skipping Test Due to location_policy Under AutoScaling in Node Pool for default-node-pool
+		// bpt.DefaultVerify(assert)
+
+		projectId := bpt.GetStringOutput("project_id")
+		location := bpt.GetStringOutput("location")
+		clusterName := bpt.GetStringOutput("cluster_name")
+		subnetName := bpt.GetStringOutput("subnet_name")
+		region := bpt.GetStringOutput("region")
+		ipRangePodsName := bpt.GetStringOutput("ip_range_pods_name")
+		ipRangeServicesName := bpt.GetStringOutput("ip_range_services_name")
+
+		op := gcloud.Runf(t, "container clusters describe %s --zone %s --project %s", clusterName, location, projectId)
+		g := golden.NewOrUpdate(t, op.String(),
+			golden.WithSanitizer(golden.StringSanitizer(projectId, "PROJECT_ID")),
+			golden.WithSanitizer(golden.StringSanitizer(clusterName, "CLUSTER_NAME")),
+		)
+		validateJSONPaths := []string{
+			"status",
+			"location",
+			"locations",
+			"privateClusterConfig.enablePrivateEndpoint",
+			"privateClusterConfig.enablePrivateNodes",
+			"privateClusterConfig.peeringName",
```

Comment:
Tests seems to be failing here. Looks like the peering name is non deterministic and we may just need to do a usual assert with the bpt.StringOutput("peering_name").

Reply:
The Test failed for Non Sanitized for Service Account. Fixed that as well